frontend/src/bundles/video_editor/components/timeline/timeline.tsx

```diff
@@ -0,0 +1,64 @@
+import { type DragEndEvent, type ItemDefinition, type Range, type ResizeEndEvent,type RowDefinition, TimelineContext  } from 'dnd-timeline';
+import { useCallback, useState } from 'react';
+
+import { TimelineView } from './subcomponents/timeline-view.js';
+
+type Properties = {
+	initialRange: Range;
+	initialRows: RowDefinition[];
+	initialItems: ItemDefinition[];
+  };
+
+  const Timeline: React.FC<Properties> = ({ initialRange, initialRows, initialItems }) => {
+
+	const [range, setRange] = useState(initialRange);
+	const [rows] = useState(initialRows);
+	const [items, setItems] = useState(initialItems);
+
+	const onResizeEnd = useCallback((event: ResizeEndEvent) => {
+		const updatedSpan =
+			event.active.data.current.getSpanFromResizeEvent?.(event);
+
+		if (!updatedSpan) {return;}
+
+		const activeItemId = event.active.id;
+
+		setItems((previous) =>
+			previous.map((item) => item.id !== activeItemId ? item : 
+				{ ...item, span: updatedSpan}
+		);
```

Comment:
I was trying to view the branch in my local setup and found some ESLint errors. `Parsing error: ')' expected.`

Reply:
My bad, I proposed this change but didn't check it with linter. I guess the parenthesis is missing after returned object: `{ ...item, span: updatedSpan})`

Reply:
The reason for that error is because of `!==` operator in ternary expression. We can solve it by changing operator to `===`:
```javascript
setItems((previous) =>
    previous.map((item) =>
        item.id === activeItemId
            ? { ...item, span: updatedSpan }
            : item,
    ),
);
```

https://eslint.org/docs/latest/rules/no-negated-condition